jupyter/d2l-java/chapter_convolutional-neural-networks/padding-and-strides.ipynb

```diff
@@ -0,0 +1,388 @@
+    "while padding with the same number of rows on top and bottom,\n",
+    "and the same number of columns on left and right.\n",
+    "\n",
+    "Moreover, this practice of using odd kernels\n",
+    "and padding to precisely preserve dimensionality\n",
+    "offers a clerical benefit.\n",
+    "For any two-dimensional array `X`,\n",
+    "when the kernels size is odd\n",
+    "and the number of padding rows and columns\n",
+    "on all sides are the same,\n",
+    "producing an output with the same height and width as the input,\n",
+    "we know that the output `Y[i, j]` is calculated\n",
+    "by cross-correlation of the input and convolution kernel\n",
+    "with the window centered on `X[i, j]`.\n",
+    "\n",
+    "In the following example, we create a two-dimensional convolutional layer\n",
+    "with a height and width of $3$\n",
+    "and apply $1$ pixel of padding on all sides.\n",
+    "Given an input with a height and width of $8$,\n",
+    "we find that the height and width of the output is also $8$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%mavenRepo snapshots https://oss.sonatype.org/content/repositories/snapshots/\n",
+    "\n",
+    "%maven ai.djl:api:0.6.0-SNAPSHOT\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "%maven net.java.dev.jna:jna:5.3.0\n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.6.0-SNAPSHOT\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ai.djl.*;\n",
+    "import ai.djl.ndarray.NDArray;\n",
+    "import ai.djl.ndarray.NDManager;\n",
+    "import ai.djl.ndarray.index.NDIndex;\n",
+    "import ai.djl.ndarray.types.Shape;\n",
+    "import ai.djl.engine.*;\n",
+    "import ai.djl.training.GradientCollector;\n",
+    "import ai.djl.nn.Block;\n",
+    "import ai.djl.nn.convolutional.Conv2D;\n",
+    "import ai.djl.training.loss.Loss;\n",
+    "import ai.djl.training.DefaultTrainingConfig;\n",
+    "import ai.djl.training.Trainer;\n",
+    "import ai.djl.training.TrainingConfig;\n",
+    "import ai.djl.training.DefaultTrainingConfig;\n",
+    "import ai.djl.nn.ParameterList;\n",
+    "import ai.djl.training.optimizer.Optimizer;\n",
+    "import ai.djl.training.optimizer.learningrate.LearningRateTracker;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load ../utils/plot-utils.ipynb\n",
+    "%load ../utils/DataPoints.java\n",
+    "%load ../utils/Training.java"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "NDArray X = manager.randomUniform(0f, 1.0f, new Shape(8, 8));\n",
```

Comment:
You can just specify the shape (1, 1, 8, 8) here instead of reshaping it on the next line